python2/raygun4py/raygunmsgs.py

```diff
@@ -135,7 +135,10 @@ def __init__(self, exc_type, exc_value, exc_traceback, options):
 
         frames = None
         try:
-            frames = inspect.getinnerframes(exc_traceback)
+            if type(exc_traceback) == list and type(exc_traceback[0]) == tuple and 'frame' in str(type(exc_traceback[0][0])).lower():
```

Comment:
This line is a bit hard to understand, could you create a method like `isStack()` which will contain all the conditional logic from this line? It would also be good to write some more tests for this as the current tests only really cover the happy-path. For example, what if `exc_traceback` is an empty list?

It would be good to go a step further and move all this logic into a separate `getFrames()` method.

Reply:
Yep, good point.